js/Search/Table.js

```diff
@@ -0,0 +1,237 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+import GenericView from './GenericView.js';
+
+// Explicitly bind to window so Jest tests work properly
+window.GLPI = window.GLPI || {};
+window.GLPI.Search = window.GLPI.Search || {};
+
+window.GLPI.Search.Table = class Table extends GenericView {
+
+   constructor(result_view_element_id) {
+      const element_id = $('#'+result_view_element_id).find('table.search-results').attr('id');
+      super(element_id);
+   }
+
+   getElement() {
+      return $('#'+this.element_id);
+   }
+
+   /**
+    * Handle sorting when a column is clicked
+    * @param target The target column element
+    * @param multisort If true, allow adding a new sort.
+    *    If false, this will only allow the user to change the sort order for the column if it is already sorted or change the sorted column.
+    */
+   onColumnSortClick(target, multisort = false) {
+      const target_column = $(target);
+      const all_colums = this.getElement().find('thead th');
+      const sort_order = target_column.attr('data-sort-order');
+
+      if (!multisort && (sort_order === null || sort_order === 'nosort')) {
+         // Remove all sorts and set this new column as the primary sort
+         all_colums.each((i, c) => {
+            const col = $(c);
+            col.attr('data-sort-num', null);
+            col.attr('data-sort-order', null);
+         });
+      }
+
+      const new_order = sort_order === 'ASC' ? 'DESC' : (sort_order === 'DESC' ? 'nosort' : 'ASC');
+      target_column.attr('data-sort-order', new_order);
+
+      let sort_num = target_column.attr('data-sort-num');
+
+      const recalulate_sort_nums = () => {
+         let sort_nums = [];
+
+         // Add sort nums to an array in order
+         all_colums.each((i, c) => {
+            const col = $(c);
+            if (col.attr('data-sort-num') !== undefined) {
+               sort_nums[col.attr('data-sort-num')] = col.attr('data-searchopt-id');
+            }
+         });
+
+         // Re-index array
+         sort_nums = sort_nums.filter(v => v);
+
+         // Convert to object and flip keys and values
+         const sort_nums_obj = {};
+         sort_nums.forEach((v, k) => {
+            sort_nums_obj[v] = k;
+         });
+
+         // Clear sort-nums from all columns or change value
+         all_colums.each((i, c) => {
+            const col = $(c);
+            col.attr('data-sort-num', undefined);
+            if (sort_nums_obj.hasOwnProperty(col.attr('data-searchopt-id'))) {
+               col.attr('data-sort-num', sort_nums_obj[col.attr('data-searchopt-id')]);
+            }
+         });
+      };
+
+      if (sort_num === undefined && new_order !== 'nosort') {
+         // Recalculate sort-num on other columns, then set new sort-num
+         recalulate_sort_nums();
```

Comment:
I do not understand why this one is necessary. Maybe it should be done on line 66.

Reply:
Moving to within that if/then block breaks sort completely. Cannot add/set any sort.